packages/gitbook/src/lib/visitors.ts

```diff
@@ -221,6 +228,28 @@ function parseVisitorQueryParamValue(value: string): ClaimPrimitive {
     return value;
 }
 
+/**
+ * Returns to cookie response to use in order to persist visitor params that were passed to the URL.
+ */
+export function getResponseCookieForVisitorParams(
+    visitorUnsignedClaims: VisitorPayloadLookup['unsignedClaims']
+): ResponseCookie | undefined {
+    if (Object.keys(visitorUnsignedClaims.fromVisitorParams).length === 0) {
+        return undefined;
+    }
+
+    return {
+        name: VISITOR_UNSIGNED_CLAIMS_PREFIX,
+        value: JSON.stringify(visitorUnsignedClaims.fromVisitorParams),
+        options: {
+            httpOnly: true,
+            sameSite: process.env.NODE_ENV === 'production' ? 'none' : undefined,
+            secure: process.env.NODE_ENV === 'production',
+            maxAge: VISITOR_PARAMS_COOKIE_MAX_AGE,
```

Comment:
Should it not be a session cookie?

Reply:
Yes good point not sure what why I did this 🤦